README.md

```diff
@@ -147,160 +157,226 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
+model = InceptionV3(weights="imagenet")
+model.save('/tmp/model-full.h5')
+```
 
-    To use the transformer, we first need to have a Keras model stored as a file. For this example we'll just save the Keras built-in InceptionV3 model instead of training one.
+Now on the prediction side:
 
-    ```python
-    from keras.applications import InceptionV3
 
-    model = InceptionV3(weights="imagenet")
-    model.save('/tmp/model-full.h5')
-    ```
+```python
+from keras.applications.inception_v3 import preprocess_input
+from keras.preprocessing.image import img_to_array, load_img
+import numpy as np
+import os
+from pyspark.sql.types import StringType
+from sparkdl import KerasImageFileTransformer
+
+def loadAndPreprocessKerasInceptionV3(uri):
+  # this is a typical way to load and prep images in keras
+  image = img_to_array(load_img(uri, target_size=(299, 299)))  # image dimensions for InceptionV3
+  image = np.expand_dims(image, axis=0)
+  return preprocess_input(image)
+
+transformer = KerasImageFileTransformer(inputCol="uri", outputCol="predictions",
+                                        modelFile='/tmp/model-full-tmp.h5',  # local file path for model
+                                        imageLoader=loadAndPreprocessKerasInceptionV3,
+                                        outputMode="vector")
+
+files = [os.path.abspath(os.path.join(dirpath, f)) for f in os.listdir("/data/myimages") if f.endswith('.jpg')]
+uri_df = sqlContext.createDataFrame(files, StringType()).toDF("uri")
+
+keras_pred_df = transformer.transform(uri_df)
+```
 
-    Now on the prediction side, we can do:
 
-    ```python
-    from keras.applications.inception_v3 import preprocess_input
-    from keras.preprocessing.image import img_to_array, load_img
-    import numpy as np
-    import os
-    from sparkdl import KerasImageFileTransformer
+#### Applying deep learning models at scale to tensors
+Deep Learning Pipelines also provides ways to apply models with tensor inputs (up to 2 dimensions), written in popular deep learning libraries:
+* TensorFlow graphs
+* Keras models
 
-    def loadAndPreprocessKerasInceptionV3(uri):
-        # this is a typical way to load and prep images in keras
-        image = img_to_array(load_img(uri, target_size=(299, 299)))
-        image = np.expand_dims(image, axis=0)
-        return preprocess_input(image)
+##### For TensorFlow users
 
-    transformer = KerasImageFileTransformer(inputCol="uri", outputCol="predictions",
-                                            modelFile="/tmp/model-full.h5",
-                                            imageLoader=loadAndPreprocessKerasInceptionV3,
-                                            outputMode="vector")
+First we generate sample dataset of 2-dimensional points, Gaussian distributed around two different centers
```

Comment:
Before diving into an example, let's add a description for TFTransformer to give more information, like:

`TFTransformer` applies a user-specified TensorFlow graph to tensor inputs of up to 2 dimensions. The TensorFlow graph may be specified as TensorFlow graph objects (`tf.Graph` or `tf.GraphDef`) or checkpoint or `SavedModel` objects (see [the input object class](https://github.com/databricks/spark-deep-learning/blob/master/python/sparkdl/graph/input.py#L27) for more detail). The `transform()` function applies the TensorFlow graph to a column of arrays (where an array corresponds to a Tensor) in the input DataFrame and outputs a column of arrays corresponding to the output of the graph.